### PR TITLE
[web][api][SIMS #1300]Removed InstitutionTypeController

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
Moved the InstitutionTypeController single method inside the institution controller.
This is the method used to load the dropdown with the institution types used on the institution profile on Institutions/AEST portals.

image

### CODE REVIEW

File: sources/packages/api/src/route-controllers/institution/institution.controller.service.ts [hunk 1]
```diff
@@ -67,4 +71,17 @@ export class InstitutionControllerService {
       hasBusinessGuid: !!institutionDetail.businessGuid,
     };
   }
+
+  /**
+   * Get the list os all institutions types to be returned in an option
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/institution/institution.controller.service.ts:76]: typo `os`

File: sources/packages/api/src/route-controllers/institution/institution.controller.service.ts [hunk 2]
```diff
@@ -67,4 +71,17 @@ export class InstitutionControllerService {
...
+   * list (key/value pair) schema.
+   * @returns institutions types in an option list (key/value pair) schema.
+   */
+  async getInstitutionTypeOptions(): Promise<OptionItemAPIOutDTO[]> {
+    const institutionTypes = await this.institutionTypeService.getAll();
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/institution/institution.controller.service.ts:81]: i know not part of the PR, but we can rename the function `getAll`

File: sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts [hunk 3]
```diff
@@ -132,4 +133,14 @@ export class InstitutionInstitutionsController extends BaseController {
       userToken.authorizations.institutionId,
     );
   }
+
+  /**
+   * Get the list os all institutions types to be returned in an option
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts:138]: typo `os`

File: sources/packages/web/src/services/InstitutionService.ts [hunk 4]
```diff
@@ -88,27 +86,32 @@ export class InstitutionService {
...
   }
 
-  public async getLocationsOptionsList(): Promise<OptionItemDto[]> {
+  async getLocationsOptionsList(): Promise<OptionItemAPIOutDTO[]> {
     return ApiClient.InstitutionLocation.getOptionsList();
   }
 
-  public async getInstitutionTypeOptions(): Promise<OptionItemDto[]> {
+  /**
+   * Get the list os all institutions types to be returned in an option
```
Comment (ann-aot) [inline sources/packages/web/src/services/InstitutionService.ts:106]: typo

File: sources/packages/api/src/route-controllers/institution/institution.controller.service.ts (hunk 2 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src/route-controllers/institution/institution.controller.service.ts:81]: Rename to what exactly?

**Review by ann-aot — COMMENTED** [review]
Looks Good. just minor comments

File: sources/packages/api/src/route-controllers/institution/institution.controller.service.ts (hunk 2 above)
Comment (ann-aot) [reply sources/packages/api/src/route-controllers/institution/institution.controller.service.ts:81]: I believe its fetching institution type. so `getInstitutionTypes` or `getAllInstitutionTypes`

File: sources/packages/api/src/route-controllers/institution/institution.controller.service.ts (hunk 1 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src/route-controllers/institution/institution.controller.service.ts]: Fixed!

File: sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts (hunk 3 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts]: Fixed!

File: sources/packages/api/src/route-controllers/institution/institution.controller.service.ts (hunk 2 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src/route-controllers/institution/institution.controller.service.ts]: Renamed!

File: sources/packages/web/src/services/http/InstitutionApi.ts [hunk 5]
```diff
@@ -74,17 +74,15 @@ export class InstitutionApi extends HttpBaseClient {
...
-        "institution/type/options-list",
-        this.addAuthHeader(),
-      );
-      return response.data;
-    } catch (error) {
-      this.handleRequestError(error);
-      throw error;
-    }
+  /**
+   * Get the list os all institutions types to be returned in an option
```
Comment (henryoforeh-aot) [inline sources/packages/web/src/services/http/InstitutionApi.ts:78]: Typo here. Is this supposed to be "Get the list of all institutions types" ?

**Review by henryoforeh-aot — APPROVED** [review]
Approved, just a minor typo fix in the comment I made.

File: sources/packages/web/src/services/http/InstitutionApi.ts (hunk 5 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/services/http/InstitutionApi.ts:78]: It was already fixed based on @ann-aot comment 😉

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @andrewsignori-aot

**Review by dheepak-aot — APPROVED** [review]
